exporter/loadbalancingexporter/metrics_exporter.go

```diff
@@ -0,0 +1,277 @@
+		exp, err = e.loadBalancer.Exporter(endpoint)
+		if err != nil {
+			return err
+		}
+
+		te, ok := exp.(exporter.Metrics)
+		if !ok {
+			return fmt.Errorf("unable to export metrics, unexpected exporter type: expected exporter.Metrics but got %T", exp)
+		}
+
+		start := time.Now()
+		err = te.ConsumeMetrics(ctx, md)
+		duration := time.Since(start)
+
+		if err == nil {
+			_ = stats.RecordWithTags(
+				ctx,
+				[]tag.Mutator{tag.Upsert(endpointTagKey, endpoint), successTrueMutator},
+				mBackendLatency.M(duration.Milliseconds()))
+		} else {
+			_ = stats.RecordWithTags(
+				ctx,
+				[]tag.Mutator{tag.Upsert(endpointTagKey, endpoint), successFalseMutator},
+				mBackendLatency.M(duration.Milliseconds()))
+		}
+	}
+	return err
+
+}
+
+func routingIdentifiersFromMetrics(mds pmetric.Metrics, key routingKey) (map[string]bool, error) {
+	ids := make(map[string]bool)
+
+	// no need to test "empty labels"
+	// no need to test "empty resources"
+
+	rs := mds.ResourceMetrics()
+	if rs.Len() == 0 {
+		return nil, errors.New("empty resource metrics")
+	}
+
+	ils := rs.At(0).ScopeMetrics()
+	if ils.Len() == 0 {
+		return nil, errors.New("empty scope metrics")
+	}
+
+	metrics := ils.At(0).Metrics()
+	if metrics.Len() == 0 {
+		return nil, errors.New("empty metrics")
+	}
+
+	for i := 0; i < rs.Len(); i++ {
+		resource := rs.At(i).Resource()
+		switch key {
+		default:
+		case svcRouting:
+			svc, ok := resource.Attributes().Get("service.name")
+			if !ok {
+				return nil, errors.New("unable to get service name")
+			}
+			ids[svc.Str()] = true
+		case metricNameRouting:
+			sm := rs.At(i).ScopeMetrics()
+			for j := 0; j < sm.Len(); j++ {
+				metrics := sm.At(j).Metrics()
+				for k := 0; k < metrics.Len(); k++ {
+					md := metrics.At(k)
+					rKey := md.Name()
+					ids[rKey] = true
+				}
+			}
+		case resourceRouting:
+			sm := rs.At(i).ScopeMetrics()
+			for j := 0; j < sm.Len(); j++ {
+				metrics := sm.At(j).Metrics()
+				for k := 0; k < metrics.Len(); k++ {
+					md := metrics.At(k)
+					rKey := resourceRoutingKey(md, resource.Attributes())
+					ids[rKey] = true
+				}
+			}
+		case attributeRouting:
+			sm := rs.At(i).ScopeMetrics()
+			for j := 0; j < sm.Len(); j++ {
+				metrics := sm.At(j).Metrics()
+				for k := 0; k < metrics.Len(); k++ {
+					md := metrics.At(k)
+					rKey := attributesRoutingKey(md, resource.Attributes())
+					ids[rKey] = true
+				}
+			}
+		}
+	}
+
+	return ids, nil
+
+}
+
+// maintain
+func sortedMapAttrs(attrs pcommon.Map) []string {
+	keys := make([]string, 0)
+	for k := range attrs.AsRaw() {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	attrsHash := make([]string, 0)
+	for _, k := range keys {
+		attrsHash = append(attrsHash, k)
+		if v, ok := attrs.Get(k); ok {
+			attrsHash = append(attrsHash, v.AsString())
+		}
+	}
+	return attrsHash
+}
+
+// @help here : what to do when metric is empty (doc says it could cause panic)
+func metricAttrsAsStr(md pmetric.Metric) []string {
```

Comment:
This would go away if we decide that routing based on metric attributes isn't desirable.